src/components/views/rooms/wysiwyg_composer/hooks/useSuggestion.ts

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Attributes, MappedSuggestion, SuggestionChar, SuggestionType } from "@matrix-org/matrix-wysiwyg";
+import { SyntheticEvent, useState } from "react";
+
+/* To allow reuse of the autocomplete component, we need to record similar information as used in 
+the Rust model (keyChar, type and text) as well as information required to allow us to do the 
+DOM manipulation when replacing the suggestion with a mention or command (node, start and
+end offsets). */
+export type PlainTextSuggestionPattern = {
+    keyChar: SuggestionChar;
+    type: SuggestionType;
+    text: string;
+    node: Node;
+    startOffset: number;
+    endOffset: number;
+};
+type Suggestion = PlainTextSuggestionPattern | null;
+
+/**
+ * React hook to allow tracking and replacing of mentions and commands in a div element
+ *
+ * @param editorRef - a ref to the div that is the composer textbox
+ * @param setText - setter function to allow the hook to update the content state when replacing
+ * text with text selected from the autocomplete
```

Comment:
Again I think this could be more explicit: 

```suggestion
 * @param setText - Setter function to set the content of the composer
```

Reply:
(likewise other references to `setText` below)